craft_application/application.py

```diff
@@ -238,13 +242,20 @@ def get_project(
         This only resolves and renders the project the first time it gets run.
         After that, it merely uses a cached project model.
 
-        :param project_dir: the base directory to traverse for finding the project file.
+        :param project_dir: (deprecated) the base directory to traverse for finding the project file.
         :param platform: the platform name listed in the build plan.
         :param build_for: the architecture to build this project for.
         :returns: A transformed, loaded project model.
         """
         if self.__project is not None:
             return self.__project
+        if project_dir is not None:
+            warnings.warn(
```

Comment:
Can this conflict with/confuse craft-cli? Ideally we should have craft-cli handling all the terminal output.

Reply:
That's a good point. I'm going to delay coming up with a solution in this particular case since @tigarmo suggested an easier way to handle this (just remove the method - https://github.com/canonical/craft-application/pull/184/files/059effce36b8c7b2bfbe7e519c26132d21df7e9c#r1482136099), but I've created https://github.com/canonical/craft-cli/issues/232 to solve this in `craft-cli`.